lib/cosmos/tools/cmd_tlm_server/replay_backend.rb

```diff
@@ -286,29 +360,80 @@ def start_playback(direction)
           @playing = false
           @playback_sleeper = nil
           @thread = nil
+          @interface.disconnect
         end
       end
     end
 
     def read_at_index(index, direction)
-      packet_offset = nil
-      packet_offset = @packet_offsets[index] if index >= 0
-      if packet_offset
-        # Read the packet
-        packet = @packet_log_reader.read_at_offset(packet_offset, false)
-        handle_packet(packet)
+      if @mode == :file
+        packet_offset = nil
+        packet_offset = @packet_offsets[index] if index >= 0
+        if packet_offset
+          # Read the packet
+          packet = @packet_log_reader.read_at_offset(packet_offset, false)
+          handle_packet(packet)
+
+          # Adjust index for next read
+          if direction == :FORWARD
+            @playback_index = index + 1
+          else
+            @playback_index = index - 1
+          end
+          @current_time_object = packet.received_time
+          @current_time = packet.received_time.formatted(true, 3, true) if packet and packet.received_time
 
-        # Adjust index for next read
-        if direction == :FORWARD
-          @playback_index = index + 1
+          return packet
         else
-          @playback_index = index - 1
+          return nil
+        end
+      else
+        unless @interface.connected?
```

Comment:
This prevents new requests from going out because the interface is already connected.

Reply:
The interface should be disconnected once a request is completed.  There shouldn't be more than one stream requested at a time per process.

Reply:
When I select a new time frame in the stream dialog it always reverts back to the first time frame and this is why I was suspecting this happened.